source/base/mpi.cc

```diff
@@ -103,6 +103,15 @@ namespace Utilities
 
 
 
+    void
+    free_communicator(MPI_Comm &mpi_communicator)
+    {
+      const int ierr = MPI_Comm_free(&mpi_communicator);
```

Comment:
I thought you were setting it to something invalid? Why not do this explicitly here.

Reply:
MPI_Comm_free does that. I added a comment.